### DISCUSSION TITLE
babel transforming not working well about recoil code

### DISCUSSION THREAD

**Discussion by toysmoon** [babel/babel#13553] [Q&A]
my project using
- next.js
- recoil

.babelrc
```json
{
  "presets": [["next/babel", { "preset-env": { "targets": "ie 11" } }]],
  "plugins": [
    "@babel/plugin-transform-parameters",
    "@babel/plugin-transform-destructuring"
  ]
}
```

when I build next.js project, I have an Error on IE 11
```
SCRIPT1010 : Expected identifier
0a301732-d3c0239a248636b1ad05.js (1,291)
```

My guess is that the code below is not being built properly.

`src/util/polyfill/recoverableViolation.js`

- before build: 
```typescript
function recoverableViolation(
  message: string,
  projectName: 'recoil',
  {error}: {error?: Error} = {},
): null {
  if (__DEV__) {
    console.error(message, error);
  }
  return null;
}
```
- after build:
```javascript
var i=function(e,t,{error:n}={}){return null};
```

after built code still have `default parameter`, `rename property`
i want transform `default parameter`, `rename property`
like this:
```
var i=function(e,t,k){var k2 = k || {}; var n = k2.error;return null};
```

**Comment by JLHwung** [comment]
Are you sure `src/util/polyfill/recoverableViolation.js` is compiled by Babel? I can not reproduce this issue on [REPL](https://babeljs.io/repl/build/47337/#?browsers=ie%2011&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=GYVwdgxgLglg9mABAJwKYTgN1cghgIwBtUA1eQ3WBACgChFEBbVAZxdwHNUAuRFqZDDAcANPUQAHZHABW6KADlczXgHI0GGIVViGAbxzTkAX14HkRgPy8AohbgnEAXkR7jYgJS8wIQoVfiMMCI1AD6oQAiNiThHgEMDBhgLHDEAHSGDtTMbJyoIoiZyB4A3OLG4mhQIMhIPn5lxkA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Ctypescript&prettier=false&targets=&version=7.14.7%2Bpr.13552&externalPlugins=). The `{error:n}` is transformed.

What is the output of `BABEL_SHOW_CONFIG_FOR="./src/util/polyfill/recoverableViolation.js" your build command`?

**Reply by toysmoon** [reply]
sorry for cofused info
`recoil/src/util/polyfill/recoverableViolation.js` is in the recoil project
I just using next.js build with `next build` command.
